types/types.go

```diff
@@ -57,6 +60,19 @@ func DefaultNodeMode() NodeMode {
 	}
 }
 
+func ParseNodeModeString(s string) NodeMode {
```

Comment:
```suggestion
func ParseNodeMode(s string) NodeMode {
```
We can already tell `s` is a string from the argument type.